lib/zuora/objects/refund.rb

```diff
@@ -1,35 +1,36 @@
-    #validates_length_of       :payment_method_id, :maximum => 60
-    #validates_datetime_of     :refund_date
-    #validates_datetime_of     :refund_transaction_time, :allow_nil => true
-    #validates_length_of       :soft_descriptor, :maximum => 35, :allow_nil => true
-    #validates_length_of       :soft_descriptor_phone, :maximum => 20, :allow_nil => true
-    #validates_inclusion_of    :source_type, :in => %w(Payment CreditBalance), :allow_nil => true
-    #validates_inclusion_of    :status, :in => %w(Canceled Error Processed Processing), :allow_nil => true
-    #validates_inclusion_of    :transferred_to_accouning, :allow_nil => true, :in => %w(Processing Yes Error Ignore)
-    #validates_inclusion_of    :type, :in => %w(Electronic External)
-    #validates_length_of       :updated_by_id, :maximum => 32
-    #validates_datetime_of     :update_date, :allow_nil => true
+    validates_length_of       :accounting_code, :maximum => 100
+    validates_numericality_of :amount
+    validates_length_of       :comment, :maximum => 255
+    validates_length_of       :created_by_id, :maximum => 32
+    # validates_datetime_of     :created_date, :allow_nil => true
+    validates_length_of       :gateway_response, :maximum => 500
+    validates_length_of       :gateway_response_code, :maximum => 20
+    validates_inclusion_of    :method_type, :in => %w(ACH Cash Check CreditCard Other PayPal WireTransfer DebitCard CreditCardReferenceTransaction)
+    validates_length_of       :payment_method_id, :maximum => 60
+    # validates_datetime_of     :refund_date
+    # validates_datetime_of     :refund_transaction_time, :allow_nil => true
+    validates_length_of       :soft_descriptor, :maximum => 35, :allow_nil => true
+    validates_length_of       :soft_descriptor_phone, :maximum => 20, :allow_nil => true
+    validates_inclusion_of    :source_type, :in => %w(Payment CreditBalance), :allow_nil => true
+    validates_inclusion_of    :status, :in => %w(Canceled Error Processed Processing), :allow_nil => true
+    validates_inclusion_of    :transferred_to_accouning, :allow_nil => true, :in => %w(Processing Yes Error Ignore)
+    validates_inclusion_of    :type, :in => %w(Electronic External)
+    validates_length_of       :updated_by_id, :maximum => 32
+    # validates_datetime_of     :update_date, :allow_nil => true
 
-    #define_attributes do
-      #read_only :accounting_code, :created_by_id, :created_date, :gateway_response,
-        #:gateway_response_code, :reference_id, :refund_number, :refund_transaction_time,
-        #:status, :updated_by_id, :updated_date
-    #end
-  #end
-#end
+    define_attributes do
+      read_only :accounting_code, :created_by_id, :created_date, :gateway_response,
+        :gateway_response_code, :reference_id, :refund_number, :refund_transaction_time,
+        :status, :updated_by_id, :updated_date
+      deferred_attributes :gateway_option_data, :payment_id, :refund_invoice_payment_data
+    end
+  end
+end
```

Comment:
And a previously commented out class, which is now uncommented. Do we know why?

Reply:
Part of the original import from upstream as seen [here](https://github.com/Sage/zuora/commit/35a6399545bd685b5054f1adcd401721e7c35190#diff-8f88ca3e66f063d39f293812a97a9dbf68ceca26a4d592d528f882b6b443e8a0).